xtest/TASKS.md

```diff
@@ -0,0 +1,313 @@
+# Test Suite Performance Optimization Tasks
+
+**Goal:** Reduce test execution time from ~30 minutes to 5-8 minutes (full run) and < 2 minutes (smoke tests)
+
+**Current Status:** 696 tests collected across test_tdfs.py and test_policytypes.py
+
+---
+
+## Phase 1: Quick Wins (1-2 days) ⚡ - ✅ COMPLETED
+
+**Results:** 3x speedup (66% reduction) with parallel execution on 14 workers
+- Sequential: 108.91s for 100 tests
+- Parallel: 36.56s for 100 tests
+
+### 1.1 Parallel Execution Setup
+- [x] Add `pytest-xdist` to requirements.txt
+- [x] Install pytest-xdist in virtual environment
+- [x] Test basic parallel execution with `pytest -n auto`
+- [x] Verify tests pass with parallel execution (all 240 smoke tests pass)
+- [x] Benchmark performance improvement
+  - **Sequential:** 107.57s for 100 smoke+roundtrip tests (FOCUS_SDK=go@main)
+  - **Parallel (14 workers):** 37.95s for same tests  
+  - **Speedup:** 2.8x faster (65% reduction in execution time)
+  - **Full smoke suite:** 240 tests pass in 167.01s with parallel execution
+
```

Comment:
Are these numbers for real?

Reply:
Yes, running locally, although this only minimal test results, not a full test suite